### PR TITLE
Intake subsystem and commands

### PR DESCRIPTION
**Pull request by OhadLevine**
this class collects the power cells

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import static frc.robot.Robot.robotConstants;
+
+/**
+ * This Class holds all the methods for the intake which collects POWER CELLs
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/Intake.java]: change POWER CELLs to POWER CELLS

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * into the robot.
+ */
+public class Intake extends SubsystemBase implements MoveableSubsystem {
+  private CANSparkMax motor;
+
+  public Intake() {
+    motor = new CANSparkMax(robotConstants.can.INTAKE_SPARK_MAX, MotorType.kBrushless);
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/Intake.java]: invert motor by a boolean robotConstant

**Review by EliyaFishman — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package frc.robot.subsystems;
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/Intake.java]: move to `frc.robot.subsystems.intake`

**Review by EliyaFishman — DISMISSED** [review]

**Comment by OhadLevine** [comment]
done

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 4]
```diff
@@ -49,5 +50,7 @@
     }
 
     // More static class here!
-
+    public static class IntakeConstants {
+        public boolean isIntakeReversed;
+    }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/constants/RobotConstants.java:75]: isIntakeReversed
to 
kReversed

**Comment by OhadLevine** [comment]
[Intake Subsystem](https://trello.com/c/ehCfTWpB/75-intake-subsystem)

File: src/main/java/frc/robot/subsystems/intake/SetIntakeSpeed.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    /**
+     * @param powerSupplier A supplier of power to be given to the intake motor
+     */
+    public SetIntakeSpeed(DoubleSupplier powerSupplier) {
+        this.powerSupplier = powerSupplier;
+    }
+
+
+    @Override
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/intake/SetIntakeSpeed.java]: }
// too many blank spaces

    @Override

File: src/main/java/frc/robot/subsystems/intake/SetIntakeSpeed.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    public SetIntakeSpeed(double power) {
+        powerSupplier = () -> power;
+    }
+
+    /**
+     * @param powerSupplier A supplier of power to be given to the intake motor
+     */
+    public SetIntakeSpeed(DoubleSupplier powerSupplier) {
+        this.powerSupplier = powerSupplier;
+    }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/intake/SetIntakeSpeed.java]: add subsystem requirement to both constractors

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
few things

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.MoveableSubsystem;
+
+import static frc.robot.Robot.robotConstants;
+
+/**
+ * This Class holds all the methods for the intake which collects POWER CELLS
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: This class

File: src/main/java/frc/robot/subsystems/intake/SetIntakeSpeed.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
+package frc.robot.subsystems.intake;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import java.util.function.DoubleSupplier;
+
+import static frc.robot.Robot.intake;
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/intake/SetIntakeSpeed.java]: too many spaces

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+ * into the robot.
+ */
+public class Intake extends SubsystemBase implements MoveableSubsystem {
+  private CANSparkMax motor;
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: motor to sparkMax

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
two things

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * This class holds all the methods for the intake which collects POWER CELLS
+ * into the robot.
+ */
+public class Intake extends SubsystemBase implements MoveableSubsystem {
+  private CANSparkMax sparkMax;
+
+  public Intake() {
+    sparkMax = new CANSparkMax(robotConstants.can.INTAKE_SPARK_MAX, MotorType.kBrushless);
+    sparkMax.setInverted(robotConstants.intakeConstants.kIntakeReversed);
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/intake/Intake.java:20]: sparkMax.setIdleMode(IdleMode.kCoast);

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
sparkMax.setIdleMode(IdleMode.kCoast);

**Comment by OhadLevine** [comment]
done, waiting renewed cr

**Review by HilelAmiel — APPROVED** [review]

**Pull request merged**